examples/gallery/plot/vectors.py

```diff
@@ -0,0 +1,65 @@
+"""
+Cartesian, circular, and geographic vectors
+-------------------------------------------
+
+The :meth:`pygmt.Figure.plot` method can plot Cartesian, circular, and geographic vectors.
+The ``style`` parameter controls vector attributes.
+
+"""
+import numpy as np
+import pygmt
+
+# create a plot with coast, Mercator projection (M) over the continental US
+fig = pygmt.Figure()
+fig.coast(
+    region=[-127, -64, 24, 53],
+    projection="M15c",
+    frame=True,
+    borders=1,
+    area_thresh=4000,
+    shorelines="0.25p,black",
+)
+
+
+# plot Cartesian vectors with different lengths
+x = np.linspace(-116, -116, 12)  # x vector coordinates
+y = np.linspace(33.5, 42.5, 12)  # y vector coordinates
+direction = np.zeros(np.shape(y))  # direction of vector
+length = np.linspace(0.5, 2.4, 12)  # length vector data
+style = "v0.2+e+a40+gred+h0+p1p,red"
+pen = "1.0p,red"
+fig.plot(x=x, y=y, style=style, pen=pen, direction=[direction, length])
+fig.text(text="CARTESIAN", x=-112, y=44.2, font="13p,Helvetica-Bold,red", fill="white")
+
+
+# plot math angle arcs with different radii
+x = -95
+y = 37
+startdir = 90  # in degrees
+stopdir = 180  # in degrees
+radius = 1.8
+pen = "1.5p,black"
+arcstyles = np.repeat("m0.5c+ea", 7)
+for arcstyle in arcstyles:
+    data = np.array([[x, y, radius, startdir, stopdir]])
+    fig.plot(data=data, style=arcstyle, color="red3", pen=pen)
+    stopdir += 40  # set the stop direction of the next circular vector
+    radius -= 0.2  # reduce radius of the next circular vector
+fig.text(text="CIRCULAR", x=-95, y=44.2, font="13p,Helvetica-Bold,black", fill="white")
```

Comment:
It would be better to prepare a 2D numpy array, and pass it to `Figure.plot()`, instead of calling `Figure.plot()` in a loop.